### ISSUE TITLE
cc_last_4 and others not being saved in db after upgrade to 1.0.0

### ISSUE THREAD

**Issue by ntherning** [killbill/killbill-stripe-plugin#8]
I had to apply the patch below to 1.0.0 to make it save `cc_last_4` and others properly. `extra_params` looks like this when `StripePaymentMethod.from_response` is called:

```ruby
{:cc_first_name=>nil, :cc_last_name=>nil, :cc_type=>nil, :cc_exp_month=>nil, :cc_exp_year=>nil, :cc_last_4=>nil}
```

The patch calls `compact` on `extra_params` to avoid those `nil` values from overriding what comes back from Stripe.

```diff
diff --git a/lib/stripe/models/payment_method.rb b/lib/stripe/models/payment_method.rb
index b07f458..6eda7e6 100644
--- a/lib/stripe/models/payment_method.rb
+++ b/lib/stripe/models/payment_method.rb
@@ -38,7 +38,7 @@ module Killbill #:nodoc:
                   :state              => card_response['address_state'],
                   :zip                => card_response['address_zip'],
                   :country            => card_response['address_country']
-              }.merge!(extra_params),
+              }.merge!(extra_params.compact), # Don't override with nil values
               model)
       end
```

**Comment by pierre** [comment]
Thanks!

Could you add a small regression test for it to make sure we catch it next time (maybe it's as simple as [adding checks in the existing tests](https://github.com/killbill/killbill-stripe-plugin/blob/master/spec/stripe/remote/integration_spec.rb#L34-L69))?

**Comment by adamlc** [comment]
Any updates on this @ntherning? Im also noticing this issue :cry:

**Comment by ntherning** [comment]
We've applied the patch. Works after that for us. I'm afraid I don't have the time to submit a test case.

**Comment by pierre** [comment]
I have released version 3.0.1 with a fix. It should be available via KPM in the next few hours.
